csrc/id_model/indexing.cpp

```diff
@@ -113,32 +93,33 @@ std::tuple<std::vector<IterDomain*>, std::vector<Val*>> getAllocationDomains(
       allocation_domains = tv->getLogicalDomain();
       contiguity = tv->domain()->contiguity();
     } else {
-      const auto inlining_pos = tv->getComputeAtPosition();
-      for (const auto i : c10::irange(tv->nDims())) {
-        auto loop_id = tv->getLoopDomain().at(i);
-        auto pt = loop_id->getParallelType();
-        if (!mayRequireAllocation(tv, loop_id)) {
-          continue;
-        }
-
-        // If the position is left of the inlining position, no need to
-        // allocate the domain unless it's shared. For example, if this
-        // is a Shared tensor and the domain is parallelized with TID,
-        // even if it's outside of the CA position, since the domain
-        // is shared, it must be allocated.
-        if (i < inlining_pos &&
-            !ir_utils::isMemorySharedAcross(tv->getMemoryType(), pt)) {
-          continue;
-        }
-
-        allocation_domains.push_back(loop_id);
-      }
+      allocation_domains = tv->getLoopDomain();
       // Assume Local and Shared are always fully contiguous
       contiguity =
           std::vector<std::optional<bool>>(allocation_domains.size(), true);
     }
   }
 
+  // Exclude some IterDomains from allocation domain
+  std::unordered_set<IterDomain*> non_allocating_ids;
+  for (const auto id : allocation_domains) {
+    auto pt = id->getParallelType();
+    if (isSizeOneDomain(id) ||
+        ir_utils::isMemoryPartitionedAcross(tv->getMemoryType(), pt)) {
+      non_allocating_ids.insert(id);
+    }
+  }
+  const auto inlining_pos = tv->getComputeAtPosition();
+  if (!tv->isFusionInput() && !tv->isFusionOutput()) {
+    for (const auto i : c10::irange(inlining_pos)) {
+      auto loop_id = tv->getLoopDomain().at(i);
+      auto pt = loop_id->getParallelType();
+      if (!ir_utils::isMemorySharedAcross(tv->getMemoryType(), pt)) {
```

Comment:
I suggest preserving the comment here.
```suggestion
      // If the position is left of the inlining position, no need to
      // allocate the domain unless it's shared. For example, if this
      // is a Shared tensor and the domain is parallelized with TID,
      // even if it's outside of the CA position, since the domain
      // is shared, it must be allocated.
      if (!ir_utils::isMemorySharedAcross(tv->getMemoryType(), pt)) {
```